exercises/1/webapp/src/components/Home.vue

```diff
@@ -74,67 +65,28 @@ export default {
     sortedNewsList: function() {
       // Sort by votes
       const compare = function (a, b) {
-  return b.votes - a.votes
+        return b.votes - a.votes
       }
       const arrayClone = Array.from(this.newsList)
       return arrayClone.sort(compare)
     }
   },
   methods: {
-      mewsUpdateMessage(newsTitle, newsVotes, voteChange){
-        const newsId = this.newsList.findIndex((item) => item.title == newsTitle)
-        if (newsId != -1) {
-          this.newsList[newsId].votes = newsVotes + voteChange
-        }
-        else {
-        }
-      },
+      newsUpdateMessage(newsTitle, newsVotes){
+        this.newsList = this.newsList.map((item) => {
+          if (item.title == newsTitle) item.votes = newsVotes
```

Comment:
You should never mutate the original elements of an array when you do `map`. This would lead to a side effect. Instead, do something like:
```suggestion
          if (item.title == newsTitle) return { ...item, votes: newsVotes }
```